spec/tom_queue/deferred_work/deferred_work_set_spec.rb

```diff
@@ -79,16 +79,6 @@
       set.pop(10).should == "work"
     end
 
-    it "should return immediately if it is interrupted by an external thread" do
-      Thread.new { sleep 0.1; set.interrupt }
-      start_time = Time.now
-      set.schedule(start_time + 1.5, "work")
-      set.schedule(start_time + 5, "work")
-      set.pop(10)
-      Time.now.should > start_time + 0.1
-      Time.now.should < start_time + 0.2
-    end
-
```

Comment:
It might be considered a hack, but you could instantiate a mixin with a replaced `exit` method, start the manager in a thread and send the thread a `SignalException` directly from the main thread - you could even publish some work to make sure the process is actually started.

Then just make sure your fake `exit` method gets called?